Db.php

```diff
@@ -194,14 +284,20 @@ static function expandRefs($value) {
      *   skip - the number of objects to skip
      *
      * @param string $collection
-     * @param array $query
-     * @param array $options
+     * @param array  $query
+     * @param array  $options
+     *
      * @return MongoCursor
      **/
-    static function find($collection, $query = array(), $options = array()) {
-        $col = self::getCollection($collection, true);
-        $fields = isset($options['fields']) ? $options['fields'] : array();
-        $result = $col->find($query, $fields);
+    public function find($collection, $query = array(), $options = array())
+    {
+        $this->readOnly = true;
```

Comment:
Do you suggest having a default value for the property `$readonly` instead of explicitly setting it within the `find` context?